r/geoarrow/R/arrow-compat.R

```diff
@@ -0,0 +1,177 @@
+
+as_arrow_array.geoarrow_vctr <- function(x, ..., type = NULL) {
+  chunked <- as_chunked_array.geoarrow_vctr(x, ..., type = type)
+  if (chunked$num_chunks == 1) {
+    chunked$chunk(0)
+  } else {
+    arrow::as_arrow_array(chunked)
+  }
+}
+
+as_chunked_array.geoarrow_vctr <- function(x, ..., type = NULL) {
+  if (is.null(type)) {
+    type <- arrow::as_data_type(attr(x, "schema", exact = TRUE))
+    chunks <- attr(x, "chunks", exact = TRUE)
+  } else {
+    stream <- as_nanoarrow_array_stream(x, schema = as_nanoarrow_schema(type))
+    chunks <- nanoarrow::collect_array_stream(stream, validate = FALSE)
+    type <- arrow::as_data_type(type)
+  }
+
+  schema <- as_nanoarrow_schema(type)
```

Comment:
We're using this conversion pattern in as_geoarrow_vctr() also. Should we extract it into a utility, or just convert x with as_geoarrow_vctr()? E.g.


```suggestion
  if (!is.null(type)) {
    array <- as_chunked_array(as_geoarrow_vctr(x, schema = as_nanoarrow_schema(type)))
    return(array)
  }

  schema <- attr(x, "schema", exact = TRUE)
  chunks <- attr(x, "chunks", exact = TRUE)
```